src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/killsProcessesWindows.groovy

```diff
@@ -0,0 +1,51 @@
+pipeline {
+    agent {
+        kubernetes {
+            customWorkspace 'c:/s'
+            yaml """
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: jnlp
+    image: "jenkins/inbound-agent:windowsservercore-1809"
+    imagePullPolicy: IfNotPresent
+    volumeMounts:
+    - mountPath: /s
+      name: s-volume
+    - mountPath: /s@tmp
+      name: stmp-volume
+    resources:
+      cpu:
+        request: 1
+      memory:
+        request: 2Gi
+  volumes:
+  - emptyDir: {}
+    name: s-volume
+  - emptyDir: {}
+    name: stmp-volume
+  nodeSelector:
+    kubernetes.io/os: windows
+"""
+        }
+    }
+    stages {
+        stage('Timeout'){
+            options {
+                timeout(time: 5, unit: "SECONDS")
+            }
+            steps{
+                container('jnlp') {
+                    script {
+                        catchError {
+                            bat 'ping 127.0.0.1 -n 3601 > test.txt' // Imagine this is a git checkout
+                        }
+                        bat 'rename test.txt test2.txt'
```

Comment:
So IIUC, the `bat` step is killed by the timeout, and then the bug was that the `ping` was not being halted and so leaving a lock on `test.txt`, causing `rename` to fail.

Reply:
Exactly that, I know is not a realistic example but is a fast and quite deterministic way to prove that processes are stopped and hence locks released.